flang/lib/Lower/Bridge.cpp

```diff
@@ -594,14 +603,18 @@ class FirConverter : public Fortran::lower::AbstractConverter {
       }
     }
     blockList.push_back(eval.nonNopSuccessor().block); // default = fallthrough
+    stmtCtx.finalize();
     builder->create<fir::SelectOp>(toLocation(), res, indexList, blockList);
   }
 
   void genFIR(const Fortran::parser::ComputedGotoStmt &stmt) {
+    Fortran::lower::StatementContext stmtCtx;
     auto &eval = getEval();
-    auto selectExpr = createFIRExpr(
-        toLocation(), Fortran::semantics::GetExpr(
-                          std::get<Fortran::parser::ScalarIntExpr>(stmt.t)));
+    auto selectExpr =
+        createFIRExpr(toLocation(),
+                      Fortran::semantics::GetExpr(
+                          std::get<Fortran::parser::ScalarIntExpr>(stmt.t)),
+                      stmtCtx);
```

Comment:
Seems like it could be a good idea.
At the moment, I just threaded the statement cleanup universally, since that is both "correct" (can't miss anything) and overly paranoid.